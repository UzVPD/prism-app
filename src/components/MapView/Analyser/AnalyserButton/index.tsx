import React from 'react';
import {
  Button,
  Typography,
  createStyles,
  withStyles,
  WithStyles,
} from '@material-ui/core';

const AnalyserButton = ({
  onClick,
  disabled,
  classes,
  label,
}: AnalyserButtonProps) => {
  return (
    <Button
      className={classes.innerAnalysisButton}
      onClick={onClick}
      disabled={disabled}
    >
      <Typography variant="body2">{label}</Typography>
    </Button>
  );
};

const styles = () =>
  createStyles({
    innerAnalysisButton: {
      backgroundColor: '#3d474a',
      margin: '10px',
      '&.Mui-disabled': { opacity: 0.5 },
    },
  });

interface AnalyserButtonProps extends WithStyles<typeof styles> {
  onClick: () => void;
  disabled?: boolean;
  label: string;
}

export default withStyles(styles)(AnalyserButton);
