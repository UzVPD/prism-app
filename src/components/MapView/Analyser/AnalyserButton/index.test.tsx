import React from 'react';
import { render } from '@testing-library/react';
import AnalyserButton from '.';

test('render AnalyserButton as expected', () => {
  const { container } = render(
    <AnalyserButton
      onClick={() => jest.fn()}
      disabled={false}
      label="Run Analysis"
    />,
  );
  expect(container).toMatchSnapshot();
});
