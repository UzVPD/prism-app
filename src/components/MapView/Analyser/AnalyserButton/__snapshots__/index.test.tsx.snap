// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AnalyserButton as expected 1`] = `
<div>
  <mock-button
    classname="AnalyserButton-innerAnalysisButton-1"
    disabled="false"
  >
    <mock-typography
      variant="body2"
    >
      Run Analysis
    </mock-typography>
  </mock-button>
</div>
`;
